File: src/main/scala/com/minalien/mffs/core/GuiHandler.scala
package com.minalien.mffs.core

import com.minalien.mffs.client.gui.{GuiCapacitor, GuiExtractor}
import com.minalien.mffs.containers.{ContainerCapacitor, ContainerForceEnergyExtractor}
import com.minalien.mffs.tiles.{TileEntityCapacitor, TileEntityExtractor}
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler

/**
 * Created by Katrina on 15/01/2015.
 */
object GuiHandler extends IGuiHandler{
  final val EXTRACTOR_GUI:Int =1
  final val CAPACITOR_GUI:Int =2
  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef =
  {
    ID match
    {
      case EXTRACTOR_GUI => new ContainerForceEnergyExtractor(player,world.getTileEntity(new BlockPos(x,y,z)).asInstanceOf[TileEntityExtractor])
      case CAPACITOR_GUI => new ContainerCapacitor(player,world.getTileEntity(new BlockPos(x,y,z)).asInstanceOf[TileEntityCapacitor])
      case _ => null
    }
  }

  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef =
  {
    ID match
    {
      case EXTRACTOR_GUI => new GuiExtractor(player,world.getTileEntity(new BlockPos(x,y,z)).asInstanceOf[TileEntityExtractor])
      case CAPACITOR_GUI => new GuiCapacitor(player,world.getTileEntity(new BlockPos(x,y,z)).asInstanceOf[TileEntityCapacitor])
      case _ => null
    }
  }
}

File: src/main/java/com/nekokittygames/mffs/common/common/GUIHandler.java
package com.nekokittygames.mffs.common.common;

import com.nekokittygames.mffs.client.gui.GuiMFFSCapacitor;
import com.nekokittygames.mffs.common.blocks.machines.MFFSCapacitor;
import com.nekokittygames.mffs.common.inventory.MFFSCapacitorContainer;
import com.nekokittygames.mffs.common.tiles.MFFSTile;
import com.nekokittygames.mffs.common.tiles.TileCapacitor;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

/**
 * Created by Katrina on 19/04/2016.
 */
public class GUIHandler implements IGuiHandler{



    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity entity=world.getTileEntity(new BlockPos(x,y,z));
        if(entity instanceof TileCapacitor)
        {
            return new MFFSCapacitorContainer(player.inventory, (MFFSTile) entity);
        }
        return null;
    }

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity entity=world.getTileEntity(new BlockPos(x,y,z));
        if(entity instanceof TileCapacitor)
        {
            return new GuiMFFSCapacitor(player.inventory, (TileCapacitor)entity,175,165,"Capacitor");
        }
        return null;
    }
}
